### PR TITLE
Feature/change match list to table

### PR DESCRIPTION
**Pull request by hansenemily**
This is a work in progress, but I want you to have a look at it since you are reluctant to use table here instead of list. For now I have just added the table in the beginning of the page, but my idea is later to remove the list completely.

I would also like to make each row expandable by clicking, and that when doing so you will get more information about the exact score of each set, referee and so on. Maybe also be able to add score/change referee from this view. Let me know what you think!

### CODE REVIEW

File: src/app/components/match-list/match-list.component.html [hunk 1]
```diff
@@ -1,3 +1,28 @@
+<table mat-table [dataSource]="setupDataSource()"  multiTemplateDataRows class="mat-elevation-z8">
```
Comment (hansenemily) [inline src/app/components/match-list/match-list.component.html:1]: I know that this way of setting up the datasource isn't pretty, but it was the only way I could get it to work. Do you have a suggestion of how I can do this in a better way? @nilssonemil

**Comment by nilssonemil** [comment]
I've taken a look at the new table, and I still have my reservations about using a table this way.

* As you mention there are more details we may want to visualize for matches, that is not required to be shown in most cases. Well-suited for an expandable item. Normally I would not expect a table row to expand. In that case it would probably be included as further columns in the table instead.
* A table normally has a quite *square* look. Including things like team logos or different styling to highlight the most important information (i.e. teams, score in contrast to date/arena) would not really fit into what I'd expect of a table. It is of course possible to do, but not the approach I'd take.
* I think it would be clearer/easier to create a match component for list elements that could be reused in the application to display information about single matches without necessarily wrapping it in a table.

Regarding styling I would suggest limiting the size of the component so that it does not fit the entire screen and reduce the amount of borders (see the Premier League exampleof unintrusive, but clear, design).

Some inspiration:

https://www.premierleague.com/results
http://elitserienvolleyboll.se/matcher-dam/ (Not super fond of the entire design, but included for variety)

Regarding

> I would also like to make each row expandable by clicking, and that when doing so you will get more information about the exact score of each set, referee and so on. Maybe also be able to add score/change referee from this view. Let me know what you think!

I think this would be a great addition!

I have yet to take a look at the actual code. Will try to do that shortly.

File: src/app/components/match-list/match-list.component.html (hunk 1 above)
Comment (nilssonemil) [reply src/app/components/match-list/match-list.component.html:1]: I think the html should refer to a member of the component, which would be initialized in ngInit which is called before the first render. I suppose this is some kind of workaround to update the data source? In this case I would probably look at https://angular.io/guide/component-interaction to find a way to communicate the data change between the this component and the one owning the data.

File: src/app/components/match-list/match-list.component.ts [hunk 2]
```diff
@@ -1,22 +1,76 @@
...
 
   getTeamNameById(id: number, teams: Team[]): string {
     return teams.find(team => team.id == id).name
   }
+
+  showReferee(id: number, teams:Team[]): string {
+    return id ? this.getTeamNameById(id, teams) : "No referee set"
+  }
+
+  getConcatinatedSets(sets: Set[]): string {
```
Comment (nilssonemil) [inline src/app/components/match-list/match-list.component.ts:51]: Maybe this could be part of the result service? Maybe could be useful elsewhere.

Comment (hansenemily) [reply src/app/components/match-list/match-list.component.ts:51]: Yes, I think that sounds like a good idea. I have been quite unsure of how you've been using the different services vs the components logic. So my approach at the moment is that if I have to use it again I will move it

File: src/app/components/match-list/match-list.component.html (hunk 1 above)
Comment (hansenemily) [reply src/app/components/match-list/match-list.component.html:1]: Yes it is a very bad workaround that I have now realized is never gonna work. Thank you for the link, hopefully I can find what I need there.

**Comment by hansenemily** [comment]
> Just a note. The API does not currently guarantee a specific order of the seasons as they are not saved with a set order.

That's true, I also realized that if you are planning ahead at the end of a season, this will also cause the wrong season to show. Will rethink this change!

**Comment by hansenemily** [comment]
> Regarding styling I would suggest limiting the size of the component so that it does not fit the entire screen and reduce the amount of borders (see the Premier League exampleof unintrusive, but clear, design).

The styling is definitely not done, I don't think it is any good yet, just wanted to get a quick draft to have something to discuss and to show you what I was thinking. The inspiration you provided is great, I had something like that in mind (though design might not be my strong suite)

**Pull request closed without merging**